File: src/main/java/com/dv/telegram/command/GetEnvironment.java
package com.dv.telegram.command;

import com.dv.telegram.WikiBot;

public class GetEnvironment extends BasicBotCommand {

    @Override
    public String getName() {
        return GetEnvironment.class.getSimpleName();
    }

    @Override
    public String getDescription() {
        return String.format(
            "%s — выдать окружение в которой запущен инстанс бота.",
            getCommandText()
        );
    }

    @Override
    public String getDefaultCommandName() {
        return "/getEnvironment";
    }

    @Override
    public String getResponse(String text, WikiBot bot) {
        return String.format(
            "%s живёт здесь: %s.",
            bot.getBotName(),
            bot.getEnvironmentName()
        );
    }
}

File: src/main/kotlin/com/dv/telegram/command/GetEnvironment.kt
package com.dv.telegram.command

import com.dv.telegram.WikiBot
import org.telegram.telegrambots.meta.api.objects.Update

class GetEnvironment : BasicBotCommand() {
    override val name: String = javaClass.simpleName

    override fun getDescription(bot: WikiBot) =
        "`${bot.botName} $commandText` — выдать окружение, в котором запущен инстанс бота."

    override val defaultCommandName = "/getEnvironment"

    override fun getResponse(text: String, bot: WikiBot, update: Update) =
        "${bot.botName} живёт здесь: ${bot.environmentName}."
}
